File: ConsoleApp1/ConsoleApp1/DauSach.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QLTV
{
	interface DacTrungQuyenSach
	{
		void MuonNgayHomNay(Date Today);
		void TraSach();
		void XDNgayTra();
	}
    abstract class DauSach
    { 
        protected string sMaDauSach;
        protected string sTua;
		protected string sNXB;
		protected string sTacGia;
		protected int iSoLuongTrongThuVien;
		protected double dGia;
		protected double dThue;
		protected double dThanhTien;

		public string MaDauSach
		{
			get { return this.sMaDauSach; }			
		}
		public string Tua
		{
			get { return this.sTua; }
		}
		public string NXB
		{
			get { return this.sNXB; }
		}
		public string TacGia
		{
			get { return this.sTacGia; }
		}
		public int SoLuongTrongThuVien
		{
			get { return this.iSoLuongTrongThuVien; }
			set { this.iSoLuongTrongThuVien = value; }
		}
		public double Gia
		{
			get { return this.dGia; }
		}
		public double Thue
		{
			get { return this.dThue; }
		}
		public double ThanhTien
		{
			get { return this.dThanhTien; }
		}
		public DauSach()
		{

		}
        public DauSach(string MaDauSach, string Tua, string NXB, string TacGia, double Gia)
        {
            this.sMaDauSach = MaDauSach;
            this.sTua = Tua;
            this.sNXB = NXB;
            this.sTacGia = TacGia;
            this.dGia = Gia;
			this.iSoLuongTrongThuVien = 0;
		}
        
		public virtual void Nhap()
		{
			Console.WriteLine(" Nhap vao ma dau sach: ");
			this.sMaDauSach = Console.ReadLine();
			Console.WriteLine(" Nhap vao tua sach: ");
			this.sTua = Console.ReadLine();
			Console.WriteLine(" Nhap vao NXB: ");
			this.sNXB = Console.ReadLine();
			Console.WriteLine(" Nhap vao tac gia: ");
			this.sTacGia = Console.ReadLine();
			Console.WriteLine(" Nhap vao gia: ");
			string Temp = Console.ReadLine();
			this.dGia= Convert.ToDouble(Temp);
			this.iSoLuongTrongThuVien = 0;
		}
		public virtual void Xuat()
		{
			
			Console.WriteLine(" Ma dau sach: " + this.sMaDauSach);
			Console.WriteLine(" Tua sach: " + this.sTua);
			Console.WriteLine(" NXB: " + this.sNXB);
			Console.WriteLine(" Tac gia: " + this.sTacGia);			
		}
		public abstract void TinhThanhTien();
		public abstract void XacDinhThue();
	}
}

File: Quan_Ly_Thu_Vien/src/DauSach.java
import java.util.Scanner;
interface DacTrungQuyenSach
{
	public void MuonNgayHomNay(Date Today);
	public void TraSach();
	public void XDNgayTra();
}
public abstract class DauSach 
{
	protected String sMaDauSach;
	protected String sTua;
	protected String sNXB;
	protected String sTacGia;
	protected int iSoLuongTrongThuVien;
	protected double dGia;
	protected double dThue;
	protected double dThanhTien;
	
	public int getiSoLuongTrongThuVien() 
	{
		return iSoLuongTrongThuVien;
	}
	public void setiSoLuongTrongThuVien(int iSoLuongTrongThuVien) 
	{
		this.iSoLuongTrongThuVien = iSoLuongTrongThuVien;
	}

	public double getdThanhTien() 
	{
		return dThanhTien;
	}
	public void setdThanhTien(double dThanhTien) 
	{
		this.dThanhTien = dThanhTien;
	}
	public String getsTenTacGia()
	{
		return this.sTacGia;
	}
	public void setsTenTacGia(String TenTacGia) 
	{
		this.sTacGia = TenTacGia;
	}
	public double getdThue() 
	{
		return this.dThue;
	}
	public void setdThue(double Tien) 
	{
		this.dThue = Tien;
	}
	public String getsMaDauSach() 
	{
		return this.sMaDauSach;
	}
	public void setsMaDauSach(String MaSach) 
	{
		this.sMaDauSach = MaSach;
	}
	public String getsNXB() 
	{
		return this.sNXB;
	}
	public void setsNXB(String NhaXuatBan) 
	{
		this.sNXB = NhaXuatBan;
	}
	public double getdGia() 
	{
		return this.dGia;
	}
	public void setdGia(double DonGia)
	{
		this.dGia = DonGia;
	}

	Scanner sScanner = new Scanner(System.in); 	
	public DauSach()
	{
		super();
	}
	public DauSach(String MaSach,String Tua,String NhaXuatBan,String TacGia, double DonGia)
	{
		super();
		this.sMaDauSach=MaSach;
		this.sTua=Tua;
		this.sNXB=NhaXuatBan;
		this.sTacGia=TacGia;
		this.dGia=DonGia;
		this.iSoLuongTrongThuVien=0;
	}
	 public void Nhap()
	 {
	        System.out.print("Nhap ma dau sach: ");
	        this.sMaDauSach = sScanner.nextLine();
	        System.out.println("Nhap tua sach: ");
	        this.sTua=sScanner.nextLine();
	        System.out.print("Nhap ten nha xuat ban: ");
	        this.sNXB = sScanner.nextLine();
	        System.out.println("Nhap tac gia: ");
	        this.sTacGia=sScanner.nextLine();
	        System.out.print("Nhap don gia: ");
	        this.dGia = sScanner.nextDouble();
	  }
	 public void Xuat()
	 {
		 	System.out.println(" Ma dau sach: "+this.sMaDauSach);
		 	System.out.println(" Tua sach: "+this.sTua);
		 	System.out.println(" NXB: "+this.sNXB);
		 	System.out.println(" Tac gia: "+this.sTacGia);
	 }
	public abstract void XacDinhThue();
	public abstract void TinhThanhTien();
}
